From 5ed25a7e5f115a71433f137905394ae6b119e450 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 7 May 2014 10:49:37 +0200
Subject: all: Remove IsDocked property

Removes the deprecated (for not very long) property, before we do a 1.0
release.

https://bugs.freedesktop.org/show_bug.cgi?id=78380

diff --git a/configure.ac b/configure.ac
index db25851..9ead3d4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,7 +43,7 @@ AC_SUBST(UP_MICRO_VERSION)
 # CURRENT	If the API or ABI interface has changed (reset REVISION to 0)
 # REVISION	If the API and ABI remains the same, but bugs are fixed.
 # AGE		Don't use.
-LT_CURRENT=2
+LT_CURRENT=3
 LT_REVISION=0
 LT_AGE=0
 AC_SUBST(LT_CURRENT)
diff --git a/libupower-glib/up-client.c b/libupower-glib/up-client.c
index 1beb220..48d99a0 100644
--- a/libupower-glib/up-client.c
+++ b/libupower-glib/up-client.c
@@ -67,7 +67,6 @@ enum {
 	PROP_ON_BATTERY,
 	PROP_LID_IS_CLOSED,
 	PROP_LID_IS_PRESENT,
-	PROP_IS_DOCKED,
 	PROP_LAST
 };
 
@@ -227,24 +226,6 @@ up_client_get_lid_is_present (UpClient *client)
 }
 
 /**
- * up_client_get_is_docked:
- * @client: a #UpClient instance.
- *
- * Get whether the machine is docked into a docking station. This property
- * is deprecated. Use XRandR, for example, to check for external displays
- * instead.
- *
- * Return value: Always %FALSE.
- *
- * Since: 0.9.2
- */
-gboolean
-up_client_get_is_docked (UpClient *client)
-{
-	return FALSE;
-}
-
-/**
  * up_client_get_on_battery:
  * @client: a #UpClient instance.
  *
@@ -339,9 +320,6 @@ up_client_get_property (GObject *object,
 	case PROP_LID_IS_PRESENT:
 		g_value_set_boolean (value, up_client_glue_get_lid_is_present (client->priv->proxy));
 		break;
-	case PROP_IS_DOCKED:
-		g_value_set_boolean (value, FALSE);
-		break;
 	default:
 		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
 		break;
@@ -418,21 +396,6 @@ up_client_class_init (UpClientClass *klass)
 							       G_PARAM_READABLE));
 
 	/**
-	 * UpClient:is-docked:
-	 *
-	 * If the laptop is docked
-	 *
-	 * Since: 0.9.8
-	 */
-	g_object_class_install_property (object_class,
-					 PROP_IS_DOCKED,
-					 g_param_spec_boolean ("is-docked",
-							       "If a laptop is docked",
-							       NULL,
-							       FALSE,
-							       G_PARAM_READABLE));
-
-	/**
 	 * UpClient::device-added:
 	 * @client: the #UpClient instance that emitted the signal
 	 * @device: the #UpDevice that was added.
diff --git a/libupower-glib/up-client.h b/libupower-glib/up-client.h
index 60bfb15..79c2d9e 100644
--- a/libupower-glib/up-client.h
+++ b/libupower-glib/up-client.h
@@ -82,7 +82,6 @@ GPtrArray	*up_client_get_devices			(UpClient		*client);
 const gchar	*up_client_get_daemon_version		(UpClient		*client);
 gboolean	 up_client_get_lid_is_closed		(UpClient		*client);
 gboolean	 up_client_get_lid_is_present		(UpClient		*client);
-gboolean	 up_client_get_is_docked		(UpClient		*client);
 gboolean	 up_client_get_on_battery		(UpClient		*client);
 
 G_END_DECLS
diff --git a/src/org.freedesktop.UPower.xml b/src/org.freedesktop.UPower.xml
index 9b2319d..c376521 100644
--- a/src/org.freedesktop.UPower.xml
+++ b/src/org.freedesktop.UPower.xml
@@ -199,20 +199,6 @@ method return sender=:1.386 -> dest=:1.451 reply_serial=2
       </doc:doc>
     </property>
 
-    <property name="IsDocked" type="b" access="read">
-      <doc:doc>
-        <doc:description>
-          <doc:para>
-            If the system is currently docked.
-            Note: the "is-docked" value is now always False.
-            Whether an external display is in use should be checked within
-            the session service applying policy on the presence or absence
-            of an external display, such as gnome-settings-daemon.
-          </doc:para>
-        </doc:description>
-      </doc:doc>
-    </property>
-
   </interface>
 
 </node>
diff --git a/src/up-daemon.c b/src/up-daemon.c
index 4a81800..c7fea62 100644
--- a/src/up-daemon.c
+++ b/src/up-daemon.c
@@ -49,7 +49,6 @@ enum
 	PROP_ON_BATTERY,
 	PROP_LID_IS_CLOSED,
 	PROP_LID_IS_PRESENT,
-	PROP_IS_DOCKED,
 	PROP_LAST
 };
 
@@ -1180,9 +1179,6 @@ up_daemon_get_property (GObject *object, guint prop_id, GValue *value, GParamSpe
 	case PROP_LID_IS_PRESENT:
 		g_value_set_boolean (value, priv->lid_is_present);
 		break;
-	case PROP_IS_DOCKED:
-		g_value_set_boolean (value, FALSE);
-		break;
 	default:
 		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
 		break;
@@ -1234,14 +1230,6 @@ up_daemon_class_init (UpDaemonClass *klass)
 							       G_PARAM_READABLE));
 
 	g_object_class_install_property (object_class,
-					 PROP_IS_DOCKED,
-					 g_param_spec_boolean ("is-docked",
-							       "Is docked",
-							       "If this computer is docked",
-							       FALSE,
-							       G_PARAM_READABLE));
-
-	g_object_class_install_property (object_class,
 					 PROP_ON_BATTERY,
 					 g_param_spec_boolean ("on-battery",
 							       "On Battery",
diff --git a/tools/up-tool.c b/tools/up-tool.c
index 69430c9..a2a35b0 100644
--- a/tools/up-tool.c
+++ b/tools/up-tool.c
@@ -136,7 +136,6 @@ up_client_print (UpClient *client)
 	g_print ("  on-battery:      %s\n", on_battery ? "yes" : "no");
 	g_print ("  lid-is-closed:   %s\n", lid_is_closed ? "yes" : "no");
 	g_print ("  lid-is-present:  %s\n", lid_is_present ? "yes" : "no");
-	g_print ("  is-docked:       no (deprecated, always false)\n");
 	action = up_client_get_critical_action (client);
 	g_print ("  critical-action: %s\n", action);
 	g_free (action);
-- 
cgit v0.10.2

